okta-spring-security-starter/src/main/java/com/okta/spring/oauth/ResourceServerConfig.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017 Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.spring.oauth;
+
+import org.springframework.beans.InvalidPropertyException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableResourceServer;
+import org.springframework.security.oauth2.config.annotation.web.configuration.ResourceServerConfigurerAdapter;
+import org.springframework.security.oauth2.config.annotation.web.configurers.ResourceServerSecurityConfigurer;
+import org.springframework.security.oauth2.provider.token.AccessTokenConverter;
+import org.springframework.security.oauth2.provider.token.DefaultTokenServices;
+import org.springframework.security.oauth2.provider.token.ResourceServerTokenServices;
+import org.springframework.security.oauth2.provider.token.TokenStore;
+import org.springframework.security.oauth2.provider.token.store.IssuerClaimVerifier;
+import org.springframework.security.oauth2.provider.token.store.JwtAccessTokenConverter;
+import org.springframework.security.oauth2.provider.token.store.JwtClaimsSetVerifier;
+import org.springframework.security.oauth2.provider.token.store.jwk.JwkTokenStore;
+import org.springframework.util.Assert;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+@EnableResourceServer
+@EnableWebSecurity
+@Configuration
+public class ResourceServerConfig extends ResourceServerConfigurerAdapter {
+
+    @Value("#{ @environment['okta.oauth.issuer'] }") //  i.e. 'https://dev-123456.oktapreview.com/oauth2/ausar5cbq5TRooicu812'
+    protected String issuerUrl;
+
+    @Value("#{ @environment['okta.oauth.audience'] ?: 'all' }")
+    protected String audience;
+
+    @Value("#{ @environment['okta.oauth.scopeClaim'] ?: 'scp' }")
+    protected String scopeClaim;
+
+    @Value("#{ @environment['okta.oauth.rolesClaim'] ?: 'groups' }")
```

Comment:
Can all these values be put in an`OktaProperties` class that can then be injected into other parts of the app? I proposed something similar for Stormpath's Java SDK: https://github.com/stormpath/stormpath-sdk-java/issues/1079